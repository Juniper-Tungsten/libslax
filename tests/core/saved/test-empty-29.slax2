version 1.2;

ns app = "http://xml.juniper.net/wonder/app";

mvar $book;
mvar $ref;

main <top> {
    set $book = {
        <book> "one";
        <book x="1"> "two";
        <book> "three";
        <book y="1"> "four";
    }
    
    for-each ($book/*[@x]) {
        <found-x> .;
    }
    
    for-each ($book/*[@y]) {
        <found-y> .;
    }
    set $book = {
        <book> "five";
        <book x="1"> "six";
        <book> "seven";
        <book y="1"> "eight";
    }
    set $ref = $book;
    set $ref = $book/*[@y];
    set $ref = $book;
    
    for-each ($ref/*) {
        <found-x> .;
    }
    
    for-each ($book/*[@y]) {
        <found-y> .;
    }
    set $book = app:get();
    set $ref = $book;
    
    for-each ($ref/*) {
        <found-x> .;
    }
    
    for-each ($book/*[@y]) {
        <found-y> .;
    }
    <book> {
        copy-of $book;
    }
    <slax-book> {
        copy-of $slax-book;
    }
}

function app:get () {
    result {
        <book> "nine";
        <book x="1"> "ten";
        <book> "eleven";
        <book y="1"> "twelve";
    }
}
