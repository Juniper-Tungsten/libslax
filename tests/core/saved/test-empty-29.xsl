<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:app="http://xml.juniper.net/wonder/app" xmlns:slax="http://xml.libslax.org/slax" xmlns:slax-func="http://exslt.org/functions" version="1.0" extension-element-prefixes="slax slax-func">
  <xsl:variable name="slax-book" mvarname="book"/>
  <xsl:variable name="book" mutable="yes" svarname="slax-book"/>
  <xsl:variable name="slax-ref" mvarname="ref"/>
  <xsl:variable name="ref" mutable="yes" svarname="slax-ref"/>
  <xsl:template match="/">
    <top>
      <slax:set-variable xmlns:slax="http://xml.libslax.org/slax" name="book" svarname="slax-book">
        <book>one</book>
        <book x="1">two</book>
        <book>three</book>
        <book y="1">four</book>
      </slax:set-variable>
      <xsl:for-each select="$book[@x]">
        <found-x>
          <xsl:value-of select="."/>
        </found-x>
      </xsl:for-each>
      <xsl:for-each select="$book[@y]">
        <found-y>
          <xsl:value-of select="."/>
        </found-y>
      </xsl:for-each>
      <slax:set-variable xmlns:slax="http://xml.libslax.org/slax" name="book" svarname="slax-book">
        <book>five</book>
        <book x="1">six</book>
        <book>seven</book>
        <book y="1">eight</book>
      </slax:set-variable>
      <slax:set-variable xmlns:slax="http://xml.libslax.org/slax" name="ref" svarname="slax-ref" select="$book"/>
      <slax:set-variable xmlns:slax="http://xml.libslax.org/slax" name="ref" svarname="slax-ref" select="$book[@y]"/>
      <slax:set-variable xmlns:slax="http://xml.libslax.org/slax" name="ref" svarname="slax-ref" select="$book"/>
      <xsl:for-each select="$ref">
        <found-x>
          <xsl:value-of select="."/>
        </found-x>
      </xsl:for-each>
      <xsl:for-each select="$book[@y]">
        <found-y>
          <xsl:value-of select="."/>
        </found-y>
      </xsl:for-each>
      <slax:set-variable xmlns:slax="http://xml.libslax.org/slax" name="book" svarname="slax-book" select="app:get()"/>
      <slax:set-variable xmlns:slax="http://xml.libslax.org/slax" name="ref" svarname="slax-ref" select="$book"/>
      <xsl:for-each select="$ref">
        <found-x>
          <xsl:value-of select="."/>
        </found-x>
      </xsl:for-each>
      <xsl:for-each select="$book[@y]">
        <found-y>
          <xsl:value-of select="."/>
        </found-y>
      </xsl:for-each>
      <book>
        <xsl:copy-of select="$book"/>
      </book>
      <slax-book>
        <xsl:copy-of select="$slax-book"/>
      </slax-book>
    </top>
  </xsl:template>
  <slax-func:function xmlns:slax-func="http://exslt.org/functions" name="app:get">
    <slax-func:result xmlns:slax-func="http://exslt.org/functions">
      <book>nine</book>
      <book x="1">ten</book>
      <book>eleven</book>
      <book y="1">twelve</book>
    </slax-func:result>
  </slax-func:function>
</xsl:stylesheet>
