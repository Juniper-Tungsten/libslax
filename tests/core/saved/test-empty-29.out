<?xml version="1.0"?>
<top xmlns:app="http://xml.juniper.net/wonder/app">
  <found-x>two</found-x>
  <found-y>four</found-y>
  <found-x>five</found-x>
  <found-x>six</found-x>
  <found-x>seven</found-x>
  <found-x>eight</found-x>
  <found-y>eight</found-y>
  <found-x>nine</found-x>
  <found-x>ten</found-x>
  <found-x>eleven</found-x>
  <found-x>twelve</found-x>
  <found-y>twelve</found-y>
  <book>
    <book>nine</book>
    <book x="1">ten</book>
    <book>eleven</book>
    <book y="1">twelve</book>
  </book>
  <slax-book>
    <book>one</book>
    <book x="1">two</book>
    <book>three</book>
    <book y="1">four</book>
    <book>five</book>
    <book x="1">six</book>
    <book>seven</book>
    <book y="1">eight</book>
    <book>nine</book>
    <book x="1">ten</book>
    <book>eleven</book>
    <book y="1">twelve</book>
  </slax-book>
</top>
